pkg/controllers/clusteroperator_controller.go

```diff
@@ -68,10 +69,127 @@ func (r *ClusterOperatorReconciler) Reconcile(ctx context.Context, _ ctrl.Reques
 	if err != nil {
 		klog.Errorf("Could not determine cluster api feature gate state: %v", err)
 		return ctrl.Result{}, r.setStatusDegraded(ctx, err)
-	} else if !capiEnabled {
-		klog.Infof("FeatureGate cluster does not include cluster api. Skipping...")
-		return ctrl.Result{}, r.setStatusAvailable(ctx)
 	}
 
+	var result ctrl.Result
+	if !capiEnabled {
+		klog.Infof("FeatureGate cluster does not include cluster api. Removing...")
+		result, err = r.reconcileDelete(ctx)
+	} else {
+		klog.Infof("FeatureGate cluster does include cluster api. Installing...")
+		result, err = r.reconcile(ctx)
+	}
+	if err != nil {
+		return result, err
+	}
+
+	return result, r.setStatusAvailable(ctx)
+}
+
+func (r *ClusterOperatorReconciler) reconcile(ctx context.Context) (ctrl.Result, error) {
+	assetNames, err := assets.AssetDir("capi-operator")
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+
+	for _, assetName := range assetNames {
+		b, err := assets.Asset(path.Join("capi-operator", assetName))
+		if err != nil {
+			return ctrl.Result{}, err
+		}
+		codecs := serializer.NewCodecFactory(r.Scheme)
+		obj, _, err := codecs.UniversalDeserializer().Decode(b, nil, nil)
+		if err != nil {
+			return ctrl.Result{}, err
+		}
+
+		if obj.GetObjectKind().GroupVersionKind().Kind == "Namespace" {
+			continue
+		}
+
+		dep, depOK := obj.(*appsv1.Deployment)
+		if depOK {
+			if err := r.customizeDeployment(dep); err != nil {
+				return ctrl.Result{}, err
+			}
+		}
+
+		existing := obj.DeepCopyObject().(client.Object)
+		_, err = ctrl.CreateOrUpdate(ctx, r.Client, existing, func() error {
+			existing = obj.(client.Object)
+			return nil
+		})
+		if err != nil {
+			return ctrl.Result{}, err
+		}
+	}
+
+	// TODO wait for the deployment to be Available?
+
+	// TODO should we install
+	// - supported provider configmaps?
+	// - infrastructure secret?
```

Comment:
All credentials will be managed by credentials requests. Similar to MAPI, we'll need a manifest that for all requests https://github.com/openshift/machine-api-operator/blob/master/install/0000_30_machine-api-operator_00_credentials-request.yaml